### PR TITLE
Primary font refactor

### PR DESCRIPTION
**Pull request by farleyawapo**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
This is an overall refactor to our use of font in the blocks library. Essentially, this should replace where possible all uses of of the `Styled Components` library and replace with the `PrimaryFont` tag from our `shared-styles` lib.

## Jira Ticket
- [TMEDIA-202](https://arcpublishing.atlassian.net/browse/TMEDIA-202)

## Acceptance Criteria
Refactor the use of styled components and where possible, take that dependency from individual block `package.json` files.
Tests should all pass.

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-202/primary_font_refactor`
2. Run fusion repo with any linked blocks: `npx fusion start -f -l @wpmedia/masthead-block`
3. Go to http://localhost/pagebuilder/editor/curate?p=pnLrGpJS6maOB88qs&v=vhtWpFUTNxaOB88qs
3. Compare blocks changed here to CoreComponents blocks page: https://corecomponents.arcpublishing.com/pagebuilder/editor/curate?p=pnLrGpJS6maOB88qs&v=vhtWpFUTNxaOB88qs
4. Check that overall font usage is correct throughout blocks, only the blocks changed need to be gone over:

Blocks to check:

- Card list block
- Date block
- Footer block
- Header block
- Masthead block
- Tag title block
- Video player block
- Video promo block
- Author Bio block
- Full author bio block
- Links bar block
- Subheadline block

## Effect Of Changes
No effects

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/card-list-block/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "@wpmedia/date-block": "canary",
     "@wpmedia/engine-theme-sdk": "canary",
     "@wpmedia/news-theme-css": "stable",
-    "styled-components": "^4.4.0"
+    "@wpmedia/shared-styles": "@wpmedia/shared-styles"
```
Comment (JackHowa) [inline blocks/card-list-block/package.json]: typo? 
```suggestion
    "@wpmedia/shared-styles": "*"
```

File: blocks/card-list-block/features/card-list/default.test.jsx [hunk 2]
```diff
@@ -276,7 +268,8 @@ describe('Card list', () => {
         expect(wrapper.find('.overline').length).toBe(0);
       });
       it('should render headline', () => {
-        expect(wrapper.find('HeadlineText.card-list-headline').length).toBe(1);
+        expect(wrapper.find('PrimaryFont.card-list-headline').length).toBe(1);
+        expect(wrapper.find('PrimaryFont.card-list-headline').length).toBe(1);
```
Comment (JackHowa) [inline blocks/card-list-block/features/card-list/default.test.jsx]: Looks like a repeat line?

```suggestion
```

File: blocks/date-block/package.json [hunk 3]
```diff
@@ -26,8 +26,7 @@
   },
   "peerDependencies": {
     "@wpmedia/engine-theme-sdk": "canary",
-    "@wpmedia/news-theme-css": "stable",
-    "styled-components": "^4.4.0"
+    "@wpmedia/news-theme-css": "stable"
   },
```
Comment (JackHowa) [inline blocks/date-block/package.json:31]: can you add shared-styles as a peer dependency?

File: blocks/date-block/features/date/default.jsx [hunk 4]
```diff
@@ -38,11 +33,11 @@ class ArticleDate extends Component {
 
   render() {
     const { displayDate } = this.state;
-    const { arcSite, classNames } = this.props;
+    const { classNames } = this.props;
     return (
-      <StyledTime key={displayDate} className={`date ${classNames}`} dateTime={displayDate} primaryFont={getThemeStyle(arcSite)['primary-font-family']}>
+      <PrimaryFont as="time" key={displayDate} className={`${classNames}`} dateTime={displayDate}>
```
Comment (JackHowa) [inline blocks/date-block/features/date/default.jsx]: looks like the classname of date is removed? is that deliberate?

File: blocks/masthead-block/features/masthead-block/masthead-block.scss [hunk 5]
```diff
@@ -1,4 +1,10 @@
 .masthead-block--text {
   font-size: calculateRem(14px);
   line-height: calculateRem(16px);
+
+}
+
+a {
+  color: #191919;
```
Comment (JackHowa) [inline blocks/masthead-block/features/masthead-block/masthead-block.scss:7]: is this being changed to a global style of any `a` tag has that color and no text decoration? I think we'd want to scope this just to masthead block styles right?

File: blocks/footer-block/features/footer/default.jsx [hunk 6]
```diff
@@ -6,21 +6,16 @@ import { useFusionContext } from 'fusion:context';
...
-
 import {
   FacebookAltIcon,
   TwitterIcon,
   RssIcon,
   LazyLoad,
   isServerSide,
 } from '@wpmedia/engine-theme-sdk';
-
+import { PrimaryFont } from '@wpmedia/shared-styles';
```
Comment (JackHowa) [inline blocks/footer-block/features/footer/default.jsx:16]: can you add shared styles as a peerDep of footer block. any block that imports shared styles should have that package as a peer dep

**Review by JackHowa — CHANGES_REQUESTED** [review]
I haven't tested this locally. Is there test instructions for the pr review? Or particular blocks that I should link? 

Looks like we're missing shared styles as a peer dep in a few places. Also, I thought the a tag affecting global styles might not be scoped as we would expect. Otherwise looks like just a few small things to change.

File: blocks/card-list-block/package.json (hunk 1 above)
Comment (farleyawapo) [reply blocks/card-list-block/package.json]: ah yes typo

File: blocks/card-list-block/features/card-list/default.test.jsx (hunk 2 above)
Comment (farleyawapo) [reply blocks/card-list-block/features/card-list/default.test.jsx]: yes repeat, will be taken out

File: blocks/date-block/package.json (hunk 3 above)
Comment (farleyawapo) [reply blocks/date-block/package.json:31]: yes, thanks for catching that

File: blocks/date-block/features/date/default.jsx (hunk 4 above)
Comment (farleyawapo) [reply blocks/date-block/features/date/default.jsx]: i was testing something and forgot to put it back, thanks for catching that

File: blocks/footer-block/features/footer/default.jsx (hunk 6 above)
Comment (farleyawapo) [reply blocks/footer-block/features/footer/default.jsx:16]: yeah i just missed a few of them, ill add that

File: blocks/masthead-block/features/masthead-block/masthead-block.scss [hunk 7]
```diff
@@ -1,4 +1,10 @@
 .masthead-block--text {
   font-size: calculateRem(14px);
   line-height: calculateRem(16px);
+
```
Comment (JackHowa) [inline blocks/masthead-block/features/masthead-block/masthead-block.scss]: this isn't working as expected?
```suggestion
   color: #191919;
   text-decoration: none;
```

File: blocks/masthead-block/features/masthead-block/masthead-block.scss (hunk 5 above)
Comment (farleyawapo) [reply blocks/masthead-block/features/masthead-block/masthead-block.scss:7]: this should be fixed now

**Review by JackHowa — CHANGES_REQUESTED** [review]
Were you able to get the shared-styles in all of the changed blocks' peer dependencies? It looks like video player block doesn't have shared-styles in the package peerDeps. also I see a few other instances of using secondaryFont the way that we were trying to refactor out. ideally we wouldn't ever see `getThemeStyle(arcSite)['secondary-font-family']` or `getThemeStyle(arcSite)['primary-font-family']` within the blocks any more


```      {description
        && (
        <DescriptionText
          secondaryFont={getThemeStyle(arcSite)['secondary-font-family']}
          className="description-text"
        >
          {description}
        </DescriptionText>
        )}```

**Comment by farleyawapo** [comment]
> Were you able to get the shared-styles in all of the changed blocks' peer dependencies? It looks like video player block doesn't have shared-styles in the package peerDeps. also I see a few other instances of using secondaryFont the way that we were trying to refactor out. ideally we wouldn't ever see `getThemeStyle(arcSite)['secondary-font-family']` or `getThemeStyle(arcSite)['primary-font-family']` within the blocks any more
> 
> ```
>         && (
>         <DescriptionText
>           secondaryFont={getThemeStyle(arcSite)['secondary-font-family']}
>           className="description-text"
>         >
>           {description}
>         </DescriptionText>
>         )}``` 
> ```

Ill add the dependency to video player block.

As far as SecondaryFont, that was not a part of this task so I did not address it in any way.

**Comment by JackHowa** [comment]
oh ok for secondary, that makes sense. yeah the including all of the peerdeps can be tricky. 

 I wasn't sure if there were other primary font calls. 

Screen Shot 2021-05-24 at 15 32 22

**Comment by farleyawapo** [comment]
> oh ok for secondary, that makes sense. yeah the including all of the peerdeps can be tricky.
> 
> I wasn't sure if there were other primary font calls.
> 

I went through all of those declarations and everything that could be refactored was refactored. since many child elements depend on `props` for the exact font, i left those alone since we wouldnt be able to pass the correct font to them without `props`

File: blocks/header-block/features/header/default.jsx [hunk 8]
```diff
@@ -1,89 +1,63 @@
...
-
-const SmallHeader = styled.h5`
-  font-family: ${(props) => props.primaryFont};
-  font-size: 20px;
-  line-height: 24px;
-`;
+import { PrimaryFont } from '@wpmedia/shared-styles';
 
 @Consumer
 class Header extends React.Component {
```
Comment (JackHowa) [inline blocks/header-block/features/header/default.jsx:8]: in the future, this could easily be a functional component but no worries

**Review by JackHowa — DISMISSED** [review]
looks good. based on the three places I spot-checked. made one change for the masthead. good job persevering through this. hope the css grows on you 

masthead block -> links have no decoration 

-> I noticed the text didn’t have the primary font fix in there. so I added it 

link decoration looks like it’s working 

all-blocks page 

header block -> font sizes for h#s match live

full author bio block -> secondary and primary fonts applied as expected

https://corecomponents.arcpublishing.com/pf/standalone-author-page/?_website=the-sun

**Review by JackHowa — APPROVED** [review]

**Pull request merged**